src/lib.rs

```diff
@@ -259,41 +254,103 @@ fn resolve_template_dir(template_base_dir: &TempDir, subfolder: Option<&str>) ->
             ));
         }
 
-        Ok(auto_locate_template_dir(
-            &template_dir,
-            prompt_for_variable,
-        )?)
+        Ok(template_dir)
     } else {
-        auto_locate_template_dir(template_base_dir.path(), prompt_for_variable)
+        Ok(template_base_dir.to_path_buf())
     }
 }
 
+/// look through the template folder structure and attempt to find a suitable template.
 fn auto_locate_template_dir(
-    template_base_dir: &Path,
-    prompt: impl Fn(&TemplateSlots) -> Result<String>,
+    template_base_dir: PathBuf,
+    prompt: &mut impl FnMut(&TemplateSlots) -> Result<String>,
 ) -> Result<PathBuf> {
-    let config_paths = locate_template_configs(template_base_dir)?;
+    let config_paths = locate_template_configs(&template_base_dir)?;
     match config_paths.len() {
-        0 => Ok(template_base_dir.to_owned()),
-        1 => Ok(template_base_dir.join(&config_paths[0])),
+        0 => {
+            // No configurations found, so this *must* be a template
+            Ok(template_base_dir)
+        }
+        1 => {
+            // A single configuration found, but it may contain multiple configured sub-templates
+            resolve_configured_sub_templates(template_base_dir.join(&config_paths[0]), prompt)
+        }
         _ => {
+            // Multiple configurations found, each in different "roots"
+            // let user select between them
             let prompt_args = TemplateSlots {
                 prompt: "Which template should be expanded?".into(),
                 var_name: "Template".into(),
                 var_info: VarInfo::String {
                     entry: Box::new(StringEntry {
-                        default: Some(config_paths[0].clone()),
-                        choices: Some(config_paths),
+                        default: Some(config_paths[0].display().to_string()),
+                        choices: Some(
+                            config_paths
+                                .into_iter()
+                                .map(|p| p.display().to_string())
```

Comment:
Maybe it's a good idea to verify here if the file or directory does exist, before asking the user?

Reply:
@sassman, not really needed, as the `locate_template_configs` function only returns paths with a `cargo-generate.toml` file in them - thus ensuring "valid" template paths.